pkg/api/index.go

```diff
@@ -109,8 +112,22 @@ func (hs *HTTPServer) getAppLinks(c *models.ReqContext) ([]*dtos.NavLink, error)
 			}
 
 			if include.Type == "dashboard" && include.AddToNav {
+				dashboardFilePath := filepath.Join(plugin.PluginDir, include.Path)
+
+				// It's safe to ignore gosec warning G304 since the variable part of the file path comes from a configuration
+				// variable.
+				// nolint:gosec
+				jsonBody, err := os.Open(dashboardFilePath)
+				if err != nil {
+					return nil, err
+				}
+				var dashboard models.Dashboard
+				jsonParser := json.NewDecoder(jsonBody)
+				if err = jsonParser.Decode(&dashboard); err != nil {
+					return nil, err
+				}
 				link := &dtos.NavLink{
-					Url:  hs.Cfg.AppSubURL + "/dashboard/db/" + include.Slug,
+					Url:  hs.Cfg.AppSubURL + "/d/" + dashboard.Uid,
```

Comment:
It doesn't feel optimal to read the dashboard.json files from disk each time requesting the index page and in addition having this kind of code in multiple places creates code duplication. 

If you could run this code when the app is installed instead you could potentially populate a new field `include.UID`. 

However, I see one potential problem with this approach and that is that it's not guarantees that a plugin dashboard would actually include a `uid` which means that Grafana generates the `uid` when the app dashboards are saved to the database.

Reply:
> If you could run this code when the app is installed instead you could potentially populate a new field include.UID.

Agree with this, but this means that we'll have to make changes to our `plugin.json` files, right?

> However, I see one potential problem with this approach and that is that it's not guarantees that a plugin dashboard would actually include a uid which means that Grafana generates the uid when the app dashboards are saved to the database.

Hm, then that means that the actual endpoints `dashboards/uid/:uid` may not work at times as well? 🤔

Reply:
@torkelo do you happen to have any thoughts on that?